Themes Showcase: Remove unused isSiteEditorActive prop

jeyip opened the pull request:
#### Description

There are no remaining `isSiteEditorActive` prop references in the themes view. This PR removes its associated selector and PropTypes validation.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Create a full site editing enabled site
* Navigate to Appearance > Themes
* Verify that the Full Site Editing Tab appears
* Remove the `fse-eligible` blog sticker from the site
* Verify that the Full Site Editing Tab disappears
* Smoke test general behavior of the themes view

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

jeyip committed:
Remove unnecessary isSiteEditorActive prop

vindl reviewed (commented):
There is one remaining usage of `isSiteEditorActive` in `shouldShowTab` that still seems relevant?

I blame GitHub for not allowing me to add an inline comment at that spot. 😄

jeyip commented:
> There is one remaining usage of isSiteEditorActive in shouldShowTab that still seems relevant?

https://github.com/Automattic/wp-calypso/pull/56916, which was recently merged to `trunk` made modifications and removed the last relevant `isSiteEditorActive` reference in `shouldShowTab`.

I rebased this PR and things should now reflect the latest version of `trunk`.

creativecoder reviewed (approved):
Looks good, no JS errors in the themes showcase (logged in or out), CI is green after triggering a rerun.
